src/Akeneo/Pim/Enrichment/Product/back/Application/UpsertProductHandler.php

```diff
@@ -33,7 +37,8 @@ public function __construct(
         private SaverInterface $productSaver,
         private ValidatorInterface $productValidator,
         private EventDispatcherInterface $eventDispatcher,
-        private UserIntentApplierRegistry $applierRegistry
+        private UserIntentApplierRegistry $applierRegistry,
+        private UserContext $userContext,
```

Comment:
```suggestion
        private UserContext $userContext,
```

Whydo we do that? UserContext is a kind of anti pattern and we should avoid its usage.

Reply:
We want to make sure the userId provided to the method is the same as the authenticatedUser.
Is it better to use this `$this->tokenStorage->getToken()->getUser()->getId()` ?

Reply:
> We want to make sure the userId provided to the method 
But why?

The user is passed by parameters, it's the responsibility of the caller to provide the user.

Reply:
As seen together, the code has been extracted with a more detailed explaination
https://github.com/akeneo/pim-community-dev/pull/17482/files#diff-9dc640bb42bdc456c174dae080c8a581ae2b8df54ac5e8ebbc4421865816f7a2R130